cmd/oras/login.go

```diff
@@ -133,24 +133,35 @@ func runLogin(opts loginOptions) (err error) {
 
 func readLine(prompt string, silent bool) (string, error) {
 	fmt.Print(prompt)
-	if silent {
-		fd := os.Stdin.Fd()
-		state, err := term.SaveState(fd)
+	fd := int(os.Stdin.Fd())
+	line := ""
+	if silent && term.IsTerminal(fd) {
+		bytes, err := term.ReadPassword(fd)
 		if err != nil {
 			return "", err
 		}
-		term.DisableEcho(fd, state)
-		defer term.RestoreTerminal(fd, state)
-	}
-
-	reader := bufio.NewReader(os.Stdin)
-	line, _, err := reader.ReadLine()
-	if err != nil {
-		return "", err
+		line = string(bytes)
+	} else {
```

Comment:
No easier way?

Reply:
Indeed. This is not relevant to term package. The aim is to implement a scanf-like function to read from **non-terminal** stdin.

Tried [reader.ReadLine](https://pkg.go.dev/bufio#Reader.ReadLine) and [fmt.Fscanf](https://pkg.go.dev/fmt#Fscanln) but both failed. The former will read extra bytes from stdin and the latter[ skips newline](https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/fmt/scan.go;l=828) before scanning starts.